src/model/delta/insertdelta.js

```diff
@@ -95,7 +104,44 @@ register( 'insert', function( position, nodes ) {
 	delta.addOperation( insert );
 	this.document.applyOperation( insert );
 
+	// When element is a DocumentFragment we need to move its markers to Document#markers.
+	if ( nodes instanceof DocumentFragment ) {
+		for ( const marker of nodes.markers ) {
+			const doc = this.document;
+			const range = moveRange( marker[ 1 ], position );
+			const markerDelta = new MarkerDelta();
+			const markerOperation = new MarkerOperation( marker[ 0 ], null, range, doc.markers, doc.version );
+
+			this.addDelta( markerDelta );
+			markerDelta.addOperation( markerOperation );
+			doc.applyOperation( markerOperation );
+		}
+	}
+
 	return this;
 } );
 
 DeltaFactory.register( InsertDelta );
+
+// Moves range relative to given position.
+//
+// @param {module:engine/model/range~Range} range
+// @param {module:engine/model/position~Position} position
+// @returns {module:engine/model/range~Range} Moved range.
+function moveRange( range, position ) {
```

Comment:
I also wonder whether one of many transformations methods in `Range` and `Position` may get handy, for example `Position#_getCombined`.